### ISSUE TITLE
Nan values for loss and accuracy in training  and testing.

### ISSUE THREAD

**Issue by adityagupta-9900** [amazon-research/siam-mot#23]
As per our capacity, we reduced it to 4 GPUs and kept the learning rate as default 0.02. After 40-60 iterations we started getting Nan value of losses
We reduced the rate to 0.015 and then trained. Even with this  for >200 iterations, sometimes it shows Nan loss values, sometimes it runs fine.
Even after testing with Nan loss values, we found all the accuracy values in the output table came out to be Nan.

![WhatsApp Image 2021-09-04 at 22 48 10](https://user-images.githubusercontent.com/42670273/132103130-f4b73a6a-5948-493e-9ce6-900c6d410130.jpeg)

**Comment by mondrasovic** [comment]
Currently, I am working on the solution to this. So far I have identified that the ground truth bounding boxes disappear during the filtering phase ([code here](https://github.com/amazon-research/siam-mot/blob/7e6ad37fd515f092704e6de89cf7cbb9f4382854/siammot/eval/eval_clears_mot.py#L39)).

As you can see on the report, everything is labeled as **FP** - **False Positive**. But why? I still have no idea, but it is one of the tasks I have to solve before moving forward.

There is another [issue](https://github.com/amazon-research/siam-mot/issues/17) with the **very same problem**.

**Comment by mondrasovic** [comment]
You can see my answer in [this issue](https://github.com/amazon-research/siam-mot/issues/17) related to this project for the **solution**.

**Comment by adityagupta-9900** [comment]
@mondrasovic  Thank you so much for your help. 
But there is another thing I needed to clarify. I'm getting Nan values even for training. Like after 40-60 iterations, I start getting Nan(Nan) loss values in training MOT data-set. 
Could you suggest why would it come like that?

**Comment by mondrasovic** [comment]
> @mondrasovic Thank you so much for your help.
> But there is another thing I needed to clarify. I'm getting Nan values even for training. Like after 40-60 iterations, I start getting Nan(Nan) loss values in training MOT data-set.
> Could you suggest why would it come like that?

I am sorry that I only answered one part of the question. I was stuck at that problem so that caused the narrow-mindedness. Nevertheless, here are my suggestions, because I also experienced the same issues.

Since I am a mortal, I do not have 8 GPUs on my local machine. As a result, I had to reduce the batch size significantly to handle the training. Well, once that happens, the learning rate should be adjusted accordingly as well. Thus, the only time when I experienced what you described with this architecture was when the learning rate produced exploding gradients. Try to observe the progress more closely, maybe you will spot a real number for a fraction of a second before it blows up to "infinity".

I would bet more than just two cents on this, because you mentioned that you used 4 GPUs and yet maintained the learning rate.

Currently, I am using
```
BASE_LR: 0.002
```
and it works.

Hope this helps!
